src/Rule.php

```diff
@@ -3553,4 +3553,163 @@ public function prepareInputForClone($input)
 
         return $input;
     }
+
+    /**
+     * Create rules (initialisation).
+     *
+     * @param boolean $reset        Whether to reset before adding new rules, defaults to true
+     * @param boolean $with_plugins Use plugins rules or not
+     * @param boolean $check        Check if rule exists before creating
+     *
+     * @return boolean
+     *
+     * @FIXME Make it final in GLPI 10.1.
+     * @FIXME Remove $reset, $with_plugins and $check parameters in GLPI 10.1, they are actually not used or have no effect where they are used.
+     */
+    public static function initRules($reset = true, $with_plugins = true, $check = false): bool
+    {
+        $self = new static();
+
+        if (!$self->hasDefaultRules()) {
+            return false;
+        }
+
+        if ($reset === true) {
+            $rules = $self->find(['sub_type' => static::class]);
+            foreach ($rules as $data) {
+                $delete = $self->delete($data);
+                if (!$delete) {
+                    return false; // Do not continue if reset failed
+                }
+            }
+
+            $check = false; // Nothing to check
+        }
+
+        $xml = simplexml_load_file(self::getDefaultRulesFilePath());
+        if ($xml === false) {
+            return false;
+        }
+
+        $ranking_increment = 0;
+        if ($reset === false) {
+            global $DB;
+            $ranking_increment = $DB->request([
+                'SELECT' => ['MAX' => 'ranking AS rank'],
+                'FROM'   => static::getTable(),
+                'WHERE'  => ['sub_type' => static::class]
+            ])->current()['rank'];
+        }
+
+        $has_errors = false;
+        foreach ($xml->xpath('/rules/rule') as $rulexml) {
+            if ((string)$rulexml->sub_type !== self::getType()) {
+                trigger_error(sprintf('Unexpected rule type for rule `%s`.', (string)$rulexml->uuid), E_USER_WARNING);
+                $has_errors = true;
+                continue;
+            }
+            if ((string)$rulexml->entities_id !== 'Root entity') {
+                trigger_error(sprintf('Unexpected entity value for rule `%s`.', (string)$rulexml->uuid), E_USER_WARNING);
+                $has_errors = true;
+                continue;
+            }
+
+            $rule = new static();
+
+            if ($check === true && $rule->getFromDBByCrit(['uuid' => (string)$rulexml->uuid])) {
```

Comment:
as we re-init the rules, could it be better to remove the old rule and re-insert it from the file ?

Reply:
`$check` param is set to `false` in our case, and does not seems to be used. I keep it to not introduce a BC break, and just reuse the same logic as previous code, but with simplier criteria.
![image](https://user-images.githubusercontent.com/33253653/213185087-fd8069a5-dd19-4b26-b167-eb434190b1e4.png)